dvc/repo/commit.py

```diff
@@ -67,5 +76,7 @@ def commit(
             filter_info=stage_info.filter_info, allow_missing=allow_missing
         )
 
-        Dvcfile(self, stage.path).dump(stage, update_pipeline=False)
+        Dvcfile(self, stage.path).dump(  # type: ignore[attr-defined]
```

Comment:
Types for `Dvcfile`/`DVCFile` are a bit odd

Reply:
yeah, I hate this too. Soon we might be able to get rid of this.